src/sth_database.js

```diff
@@ -8,53 +8,137 @@
   var sthConfig, sthLogger, sthHelper, connectionURL, eventSchema, aggregatedSchema;
 
   /**
+   * Declares the Mongoose schemas.
+   */
+  function defineSchemas() {
+    switch (sthConfig.DATA_MODEL) {
+      case sthConfig.DATA_MODELS.COLLECTIONS_PER_SERVICE_PATH:
+        eventSchema = mongoose.Schema({
+          timestamp: Date,
+          entityId: String,
+          entityType: String,
+          attributeName: String,
+          attributeType: String,
+          attributeValue: Number
+        });
+        break;
+      case sthConfig.DATA_MODELS.COLLECTIONS_PER_ENTITY:
+        eventSchema = mongoose.Schema({
+          timestamp: Date,
+          attributeName: String,
+          attributeType: String,
+          attributeValue: Number
+        });
+        break;
+      case sthConfig.DATA_MODELS.COLLECTIONS_PER_ATTRIBUTE:
+        eventSchema = mongoose.Schema({
+          timestamp: Date,
+          attributeType: String,
+          attributeValue: Number
+        });
+        break;
+    }
+
+    switch (sthConfig.DATA_MODEL) {
+      case sthConfig.DATA_MODELS.COLLECTIONS_PER_SERVICE_PATH:
+        aggregatedSchema = mongoose.Schema({
+          _id: {
+            type: {
+              entityId: String,
+              entityType: String,
+              attributeName: String,
+              range: String,
+              resolution: String,
+              origin: Date
+            },
+            select: true
+          },
+          attributeType: String,
+          points: [{
+            offset: Number,
+            samples: Number,
+            sum: Number,
+            sum2: Number,
+            min: Number,
+            max: Number
+          }]
+        });
+        break;
+      case sthConfig.DATA_MODELS.COLLECTIONS_PER_ENTITY:
+        aggregatedSchema = mongoose.Schema({
+          _id: {
+            type: {
+              attributeName: String,
```

Comment:
In fact, fixed in https://github.com/telefonicaid/IoT-STH/commit/d45f39bd4cd9d62980a1d3bef5cec0adb5fe09c6